cmake/Subdirectories.cmake

```diff
@@ -255,27 +346,42 @@ function(get_all_external_subdirectories_for_o3de_object output_subdirs object_t
     # These gems are registered in the users o3de_manifest.json
     o3de_read_json_array(initial_gem_names ${object_path}/${object_json_filename} "gem_names")
     set(gem_names "")
-    foreach(gem_name IN LISTS initial_gem_names)
+
+    # Gem dependency resolution can be disabled to speed up configuration 
+    # for projects where it is not needed
+    if(initial_gem_names AND NOT O3DE_DISABLE_GEM_DEPENDENCY_RESOLUTION)
+        # Resolve gem dependency names to gem paths before adding them to external subdirectories 
+        resolve_gem_dependencies(${object_type} "${object_path}")
+    endif()
+
+    foreach(gem_name_with_version_specifier IN LISTS initial_gem_names)
+
         # Use the ERROR_VARIABLE to catch the common case when it's a simple string and not a json type.
-        string(JSON json_type ERROR_VARIABLE json_error TYPE ${gem_name})
+        string(JSON json_type ERROR_VARIABLE json_error TYPE ${gem_name_with_version_specifier})
         set(gem_optional FALSE)
         if(${json_type} STREQUAL "OBJECT")
-            string(JSON gem_optional GET ${gem_name} "optional")
-            string(JSON gem_name GET ${gem_name} "name")
+            string(JSON gem_optional GET ${gem_name_with_version_specifier} "optional")
+            string(JSON gem_name_with_version_specifier GET ${gem_name_with_version_specifier} "name")
         endif()
 
+        # Remove any version specifier from the gem name
+        o3de_get_name_and_version_specifier(${gem_name_with_version_specifier} gem_name spec_op spec_version)
+
         # Set a global "optional" property on the gem name
         set_property(GLOBAL PROPERTY "${gem_name}_OPTIONAL" ${gem_optional})
+
         # Build the gem_names list with extracted names
-        list(APPEND gem_names ${gem_name})
+        list(APPEND gem_names ${gem_name_with_version_specifier})
```

Comment:
After this loop a call to `ly_enable_gems(GEM ${gem_name} PROJECT_NAME ${object_name}` can be added to make sure the gems referenced in the "gem_name" field of a project.json or engine.json is added to the generated `cmake_dependencies.*.setreg` for loading